Tests/AmplitudeTests.m

```diff
@@ -1549,4 +1549,160 @@ - (void)testRateLimitBackoffLogic {
+                                                             screenViews:YES];
+    [client initializeApiKey:@"default_tracking"];
+
+    XCTAssertFalse(client.defaultTracking.sessions);
+    XCTAssertTrue(client.defaultTracking.appLifecycles);
+    XCTAssertTrue(client.defaultTracking.screenViews);
+    XCTAssertTrue(client.defaultTracking.deepLinks);
+}
+
+
+#if !TARGET_OS_OSX && !TARGET_OS_WATCH
+- (void)testObserveDidFinishLaunchingNotification {
+    id mockApplication = [OCMockObject niceMockForClass:[UIApplication class]];
+    [[[mockApplication stub] andReturn:mockApplication] sharedApplication];
+    OCMStub([mockApplication applicationState]).andReturn(UIApplicationStateInactive);
+    
+    NSString *instanceName = @"default_tracking_ObserveDidFinishLaunchingNotification";
+    Amplitude *client = [Amplitude instanceWithName:instanceName];
+    client.defaultTracking.appLifecycles = YES;
+    [client initializeApiKey:@"default_tracking"];
+
+    UIApplication *app = [AMPUtils getSharedApplication];
+    NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
+    [center postNotificationName:UIApplicationDidFinishLaunchingNotification object:app];
+    
+    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper:instanceName];
+    [dbHelper insertOrReplaceKeyValue:@"app_build" value:nil];
+    [dbHelper insertOrReplaceKeyValue:@"app_version" value:nil];
+
+    [client flushQueue];
+
+    NSDictionary *event1 = [client getLastEventFromInstanceName:instanceName fromEnd: 1];
+    XCTAssertEqualObjects([event1 objectForKey:@"event_type"], @"[Amplitude] Application Installed");
+
+    NSDictionary *event2 = [client getLastEventFromInstanceName:instanceName fromEnd: 0];
+    XCTAssertEqualObjects([event2 objectForKey:@"event_type"], @"[Amplitude] Application Opened");
+    XCTAssertEqualObjects([[event2 objectForKey:@"event_properties"] objectForKey:@"[Amplitude] From Background"], @NO);
+}
+
+- (void)testObserveDidFinishLaunchingNotificationWithPreviousBuild {
+    id mockApplication = [OCMockObject niceMockForClass:[UIApplication class]];
+    [[[mockApplication stub] andReturn:mockApplication] sharedApplication];
+    OCMStub([mockApplication applicationState]).andReturn(UIApplicationStateInactive);
+    
+    NSString *instanceName = @"default_tracking_ObserveDidFinishLaunchingNotificationWithPreviousBuild";
+    Amplitude *client = [Amplitude instanceWithName:instanceName];
+    client.defaultTracking.appLifecycles = YES;
+    [client initializeApiKey:@"default_tracking"];
+
+    UIApplication *app = [AMPUtils getSharedApplication];
+    NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
+    [center postNotificationName:UIApplicationDidFinishLaunchingNotification object:app];
+    
+    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper:instanceName];
+    [dbHelper insertOrReplaceKeyValue:@"app_build" value:@"test"];
```

Comment:
Question:
Where did we set the currentBuild and current Version?

Reply:
Good question, I put more details in the [doc](https://amplitude.atlassian.net/wiki/spaces/GROWTH/pages/2109079759/Test+Mobile+SDK+Default+Events#Application-Updated.1). This can be set in the Xcode project target settings.